src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -923,9 +923,22 @@ public override async Task WriteAsync(byte[] buffer, int offset, int count, Canc
                     throw new InvalidDataException(SR.GenericInvalidData);
```

Comment:
Technically there's a breaking change here, in that we previously were returning any exception from here or from EnsureNotDisposed out through the returned task.  But in practice, in this case I don't think it really matters.  The only caller of this is going to be the source stream's CopyToAsync, and it seems really unlikely that it'll a) call this overload (the base implementation calls the memory overload), b) misuse the method (e.g. call it when disposed), AND c) do something other than directly await the call's result, in which case whether an exception is propagated directly or via the returned task isn't perceivable.  On top of that, there's a good argument to be made that at least the EnsureNotDisposed shouldn't have been wrapped in the first place.  But if we wanted to maintain compat for this super rare case, we could wrap these with Task.FromException.

Reply:
Just to clarify, do you want me to leave it as-is?

Reply:
Let's leave the EnsureNotDisposed as is, but change this line to be:
```C#
return Task.FromException(new InvalidDataException(SR.GenericInvalidData));
```